KarishmaThaladi_final_assessment.py

```diff
@@ -0,0 +1,211 @@
+        return '{} {}'.format(self.name, self.surname)
+
+
+class NanoStudent(CFGStudent):
+
+    def __init__(self, specialization, name, surname, age, email, course_grades=None, students=None ):
+        super().__init__(name, surname, age, email)
+        self.specialization = specialization
+
+        if students is None:
+            self.students = []
+        else:
+            self.students = students
+
+        if course_grades is None:
+            self.course_grades = []
+        else:
+            self.course_grades = course_grades
+
+    @staticmethod
+    def generate_id():
+        'your code goes here'
+        'create a random new id, which is a word NANO followed by any number between 1,000 and 10,000'
+        'return id as a string'
+        "Example 'NANO1234' "
+        return 'NANO'+str(random.randint(1000, 10000))
+
+    def add_new_grade(self, new_grade):
+        'your code goes here'
+        for course_grades in self.course_grades:
+            self.new_grade = new_grade
+        'update course_grades container'
+        "Example: pass in a task name and its grade, so that both are added to course_grades"
+
+    def get_course_grades(self):
+        'your code goes here'
+        'fetch course grades container and its values'
+        for student in self.students:
+            print('--> ' + student.name() + student.surname() + '--> ' + student.course_grades)
+
+
+############################################
+
+# Example run
+
+s = CFGStudent('Anna', 'Smith', 18, 'anna@mail.com')  # do not pass ID, it should be generated automatically
+print(s.get_fullname())
+# returns 'Anna Smith'
+print(s.student_id)
+# returns '3868' or some other random number
+
+cfg_s = NanoStudent('Software', name='Mia', surname='Papandopulu', age=20, email='mia@mail.com')
+print(cfg_s.get_fullname())
+# returns 'Mia Papandopulu'
+print(cfg_s.get_id())
+# returns 'NANO6180' or some other random number
+
+cfg_s.add_new_grade('theory', 95)
+cfg_s.add_new_grade('project', 78)
+print(cfg_s.get_course_grades())
+# returns {'theory': 95, 'project': 78}
+
+
+
+"""
+TASK 2
+
+Given an index limit, find all corresponding Fibonacci values up to that limit in a sequence 
+and return the sum of all even Fibonacci numbers. See more details in the task description in 
+your assessment paper. 
+"""
+def even_fibonacci_sum(limit):
+    #Fibonacci series
+    fib = [0, 1]  # starting from 0 and 1
+    for i in range(2, limit + 1):
+        fib.append(fib[-1] + fib[-2])
+    print('Fibonacci series: ', fib)
+
+    #Even fiboancci series
+    even = []
+    for x in fib:
+        if x % 2 == 0:
+            even.append(x)
+    print('Fibonacci even series: ', even)
+
+    #Sum of even numbers
+    print("Sum of even fibonacci series: ", sum(even))
+
+
+##### TESTS ####
+
+print(even_fibonacci_sum(limit=10))  # should be 44
+print(even_fibonacci_sum(limit=15))  # should be 188
+print(even_fibonacci_sum(limit=1))   # should be 0
+

```

Comment:
Good solution, except the limit in your function is inclusive, meaning that your function goes through 0-15, which are 16 numbers in total. The limit range should have been excluding the limit itself. If you accounted for that everything would have been perfect